libs/global-contract/contract/src/lib.rs

```diff
@@ -66,15 +69,95 @@ impl GlobalData {
 
 #[near_bindgen]
 impl CommunityContract for GlobalData {
-    fn set_community_contract(&mut self, server: String, contract_address: AccountId) {
-        if !(self.admins.contains(&predecessor_account_id())) {
+    // fn set_community_contract(&mut self, server: String, contract_address: AccountId) {
+    //     if !(self.admins.contains(&predecessor_account_id())) {
+    //         throw_error(crate::errors::UNAUTHORIZED);
+    //     }
+    //     self.guild_id_to_community_contract
+    //         .insert(&server, &contract_address);
+    // }
+    // fn get_community_contract(&self, server: String) -> Option<AccountId> {
+    //     self.guild_id_to_community_contract.get(&server)
+    // }
+
+    fn init_community(&mut self, guild_id: String, new_admins: Vec<AccountId>) {
+        if !self.admins.contains(&predecessor_account_id()) {
+            throw_error(crate::errors::UNAUTHORIZED);
+        }
+        let mut admins = UnorderedSet::new(format!("community-contract-{}", guild_id).as_bytes());
+        for new_admin in new_admins.iter() {
+            admins.insert(new_admin);
+        }
+        self.guild_id_to_community_info.insert(
+            &guild_id,
+            &CommunityInfo {
+                admins,
+                default_nft_contract: None,
+            },
+        );
+    }
+
+    fn set_community_default_nft_contract(&mut self, guild_id: String, nft_contract: AccountId) {
+        let community_opts = self.get_community_info(&guild_id);
+        if community_opts.is_none() {
+            throw_error(crate::errors::GUILD_ID_NOT_REGISTERED);
+        };
+        let mut community = community_opts.unwrap();
+        let admins = community.admins;
+        if !admins.contains(&predecessor_account_id()) {
             throw_error(crate::errors::UNAUTHORIZED);
         }
-        self.server_to_community_contract
-            .insert(&server, &contract_address);
+        community.default_nft_contract = Some(nft_contract);
+        community.admins = admins;
+        self.guild_id_to_community_info
+            .insert(&guild_id, &community);
+    }
+
+    fn get_community_default_nft_contract(&self, guild_id: String) -> Option<AccountId> {
+        let community_opts = self.get_community_info(&guild_id);
```

Comment:
same as above, `match` instead of `unwrap`

Reply:
I changed to have it throw error within get_community_info